src/main/java/com/firebolt/jdbc/client/query/StatementClientImpl.java

```diff
@@ -153,30 +180,47 @@ public void abortStatement(String id, FireboltProperties fireboltProperties) thr
 				throw e;
 			}
 		} catch (Exception e) {
-			throw new FireboltException(format("Could not cancel query: %s at %s", id, fireboltProperties.getHost()), e);
+			throw new FireboltException(format("Could not cancel query: %s at %s", label, fireboltProperties.getHost()), e);
+		}
+	}
+
+	@SuppressWarnings("java:S2142") // "InterruptedException" and "ThreadDeath" should not be ignored
+	private void delay(int delay) {
+		try {
+			Thread.sleep(delay);
+		} catch (InterruptedException e) {
+			// ignore interrupted exception
 		}
 	}
 
 	/**
 	 * Abort HttpRequest if it is currently being sent
 	 *
-	 * @param id id of the statement
+	 * @param label id of the statement
 	 */
-	public void abortRunningHttpRequest(@NonNull String id) {
-		getQueuedCallWithId(id).ifPresent(Call::cancel);
-		getRunningCallWithId(id).ifPresent(Call::cancel);
+	private boolean abortRunningHttpRequest(@NonNull String label) {
+		boolean quedAborted = abortCallWithId(getQueuedCallWithId(label));
+		boolean runningAborted = abortCallWithId(getRunningCallWithId(label));
+		return quedAborted || runningAborted;
+	}
+
+	private boolean abortCallWithId(Optional<Call> call) {
```

Comment:
nit: `abortCallWithId` -> `abortCall` since no id is passed

Reply:
renamed